### PR TITLE
Add Authenticated Users support

### PR DESCRIPTION
**Pull request by luismfonseca**
This implements the `generateToken` and `publishToUsers` functions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
```
Comment (lukabratos) [inline CHANGELOG.md]: I would put a link here which would show a diff between the latest release and new commits:
For example, if `1.0.0` is the latest release: 
`https://github.com/pusher/push-notifications-server-java/compare/1.0.0...HEAD`

Comment (luismfonseca) [reply CHANGELOG.md]: Ok, I'll do it outside this PR!

File: src/main/kotlin/com/pusher/pushnotifications/PushNotifications.kt [hunk 2]
```diff
@@ -39,22 +51,70 @@ class PushNotifications(private val instanceId: String, private val secretKey: S
...
      */
+    @Deprecated("use publishToInterests instead", ReplaceWith("publishToInterests(interests, publishRequest)"))
     @Throws(IOException::class, InterruptedException::class, URISyntaxException::class)
     fun publish(interests: List<String>, publishRequest: Map<String, Any>): String {
-        this.validateInterests(interests)
+        return publishToInterests(interests, publishRequest)
+    }
+
+    /**
+     * Publish the given publish_body to the specified interests.
```
Comment (jonathanlloyd) [inline src/main/kotlin/com/pusher/pushnotifications/PushNotifications.kt]:
```suggestion
     * Publish the given publishRequest to the specified interests.
```

File: src/main/kotlin/com/pusher/pushnotifications/PushNotifications.kt [hunk 3]
```diff
@@ -39,22 +51,70 @@ class PushNotifications(private val instanceId: String, private val secretKey: S
...
+            throw IllegalArgumentException("interest $it is longer than the maximum of $interestNameMaxLength characters")
+        }
 
         val publishRequestWithInterests = publishRequest.toMutableMap()
         publishRequestWithInterests.put("interests", interests)
 
         val client = HttpClients.createDefault()
-        val httpPost = HttpPost("$baseURL/instances/$instanceId/publishes")
+        val httpPost = HttpPost("$baseURL/publish_api/v1/instances/$instanceId/publishes/interests")
         httpPost.setEntity(StringEntity(gson.toJson(publishRequestWithInterests)))
```
Comment (jonathanlloyd) [inline src/main/kotlin/com/pusher/pushnotifications/PushNotifications.kt]: Request logic could be abstracted

**Pull request merged**